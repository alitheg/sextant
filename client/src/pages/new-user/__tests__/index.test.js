// Link.react.test.js
import React from 'react';
import NewUserPage from '..';
import {
  BrowserRouter as Router
} from "react-router-dom";
import renderer from 'react-test-renderer';
import {cleanup, render, fireEvent} from '@testing-library/react';
import faker from 'faker'

const api = require('../../../api');

var mockPostHandler = {};
// const getMockPostHandler = () => mockPostHandler
jest.mock('../../../api', () => ({
  usePostHandler: jest.fn().mockImplementation(() => {
    return mockPostHandler
  })
}));

afterEach(cleanup);

describe('New User Page', () => {
  test('Matches snapshots', () => {
    let component = renderer.create(
      <Router>
        <NewUserPage loggedIn={false} />
      </Router>
    ); // TODO: update form to redirect to home page

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
      <Router>
        <NewUserPage loggedIn={true} />
      </Router>
    );

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  // test('Create button calls API', () => {
  //   const updateData = jest.fn()
  //   const username = faker.internet.userName();
  //   const password = faker.internet.password();
  //   mockPostHandler = {data: {}, handleRequest: updateData}
  //   const { getByLabelText, getByText } = render(
  //     <Router>
  //       <NewUserPage />
  //     </Router>
  //   );
  //
  //   const usernameField = getByLabelText(/Username/i).closest('input');
  //   usernameField.value = username;
  //   fireEvent.change(usernameField, {target: {value: username}})
  //   const passwordField = getByLabelText(/Password/i).closest('input');
  //   passwordField.value = password;
  //   fireEvent.change(passwordField, {target: {value: password}})
  //   getByText(/Create/i).closest('button').click()
  //
  //   expect(updateData).toHaveBeenCalledTimes(1)
  //   // expect(login).toHaveBeenCalledWith('/api/login', {username, password}) // TODO: onChange events aren't firing
  // })
})
