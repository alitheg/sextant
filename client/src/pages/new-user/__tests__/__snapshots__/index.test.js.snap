// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New User Page Matches snapshots 1`] = `
<div
  className="container-fluid"
>
  <div>
    <h2>
      New user
    </h2>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="userUpdate.Username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="string"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="userUpdate.Password"
          onChange={[Function]}
          pattern="(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])\\\\S{6,}"
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Create
      </button>
    </form>
  </div>
  Error
</div>
`;

exports[`New User Page Matches snapshots 2`] = `
<div
  className="container-fluid"
>
  <div>
    <h2>
      New user
    </h2>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="userUpdate.Username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="string"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="userUpdate.Password"
          onChange={[Function]}
          pattern="(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])\\\\S{6,}"
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Create
      </button>
    </form>
  </div>
  Error
</div>
`;
